integtest/src/main/java/org/corfudb/universe/cluster/docker/DockerCluster.java

```diff
@@ -0,0 +1,136 @@
+package org.corfudb.universe.cluster.docker;
+
+import com.google.common.collect.ImmutableList;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.NetworkConfig;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.universe.cluster.Cluster;
+import org.corfudb.universe.cluster.ClusterException;
+import org.corfudb.universe.service.DockerService;
+import org.corfudb.universe.service.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static lombok.Builder.Default;
+
+/**
+ * Represents Docker implementation of a {@link Cluster}.
+ */
+@Slf4j
+@Builder
+@AllArgsConstructor
+public class DockerCluster implements Cluster {
+
+    @Getter
+    private final ClusterParams clusterParams;
+    private final DockerClient docker;
+    private final DockerNetwork network = new DockerNetwork();
+    @Default
+    private final ImmutableList<Service> services = ImmutableList.of();
+
+    /**
+     * Deploy a cluster according to provided cluster parameter, docker client, docker network, and other cluster
+     * components.
+     * The instances of this class are immutable. In other word, when the state of an instance is changed a new
+     * immutable instance is provided.
+     *
+     * @return New immutable instance of docker cluster would be returned.
+     * @throws ClusterException this exception will be thrown if deploying a cluster is not successful
+     */
+    @Override
+    public DockerCluster deploy() throws ClusterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.corfudb.universe.cluster.ClusterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)